### DISCUSSION TITLE
Google Batch prediction

### DISCUSSION THREAD

**Discussion by hect1c** [langchain-ai/langchain-google#599] [Announcements]
From my understanding, please correct me if I'm wrong, when using Langchain's batch method we are essentially just running Runnables.batch which appears to just run the invoke method but in parallel using threadpools or async tasks. 

So my questions are:

1. Is the invocation method for Langchain Vertex AI API the same regardless of which method we choose (batch vs invoke)?
2. Under the hood for Vertex is Langchain using the predict endpoint for either invocation method?
3.  If above is true, are there plans to implement batch prediction API with Vertex?

I would like to use GCPs Batch prediction with Vertex AI while leveraging the functionality and features / tools from Langchain. Is there a way to achieve this?

**Comment by ACMCMC** [comment]
I'd be interested in this as well. I'd like to be able to use a batch prediction endpoint (https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/batch-prediction-gemini) instead of the regular text generation one, on a batched input.

**Comment by jmorenobrasero** [comment]
@hect1c If you want to do so you will need to use native library and specify all the inputs in a file stored in GCS. This could be done on Langchain but not sure how useful is to do it within Langchain

**Reply by PoorRican** [reply]
Langchain provides a unified interface for defining messages & tools. When running evaluations that will use multiple providers, it's much easier to construct everything using Langchain's `BaseTool`, `HumanMessage`, etc, and then use builtin mechanisms to get the JSONL required for uploading to the batch API.

With other providers, its simpler, as they accept base64 encoded images. Vertex AI requires you to use a bucket for some reason...
